### ISSUE TITLE
Imported Images opened in darkroom mode and closed show as "Edited" even if no changes are done (DT 3.0RC3)

### ISSUE THREAD

**Issue by arigit** [darktable-org/darktable#3815]
In the past (DT 3.0RC1 and prior stable releases), importing an image (RAW or even JPG for culling), entering the darkroom to inspect it and returning to the lighttable would NOT cause the "edited" flag (ying-yang symbol) to turn on if no changes were made in the darkroom.

Also, in the darkroom it was possible to "compress" the history stack back to the Original Image and discard everything else - this was a way to revert any editing (in addition to using "Discard History Stack" in the lighttable)

WIth RC3, any attempt to "compress" the history stack reverts to including nine (!) items in the stack, and there is no way to get the "Edited" flag to turn off (unless using "Discard History Stack" in the lighttable)

![image](https://user-images.githubusercontent.com/652673/71192374-21ee0400-2256-11ea-81a3-822874bbf830.png)

This seems to be a bug. In prior releases, compressing the stack down to "Original Image" and exiting the darkroom would resent the Edited flag, and upon entering the darkroom again, the "autoapplied" modules would pop up again (just baseline and orientation) but this would NOT cause the Edited flag to turn on. 

Even JPGs that are visualized in the darkroom (e.g. just to check focus / zoom) are flagged as edited upon exiting the darkroom even if not changes are done whatsoever.

Any chance to revert to the prior behavior for the Edited flag that was showing if any **real** editing was done or not to the picture?

**Comment by TurboGit** [comment]
Can you test with #3817 ?

Quite urgent if we want this in 3.0 ! Thanks.

**Comment by arigit** [comment]
Thanks for the fast response
Built your branch, installed and tested: it works fine. With this patch, entering the darkroom and exiting the darkroom without modifying the image no longer triggers the Edited flag. Also, doing any editing in the image turns on the flag as expected.
Lastly, entering the darkroom with an edited image, and rolling back the history stack by compressing it back to the "Original Image", and then exiting the image, clears the flag.

Bottom line the patch restored the correct and expected behavior. Please merge

**Comment by homer3018** [comment]
@arigit For the sake of checking, have you had time to try also to preview from the lighttable a non edited picture with the (default) preview key, W ? To ensure a full size rendering also does not trigger the flag ? Thanks !

**Comment by arigit** [comment]
@homer3018 tested - with the patch, pressing W does not cause the picture to be flagged as Edited. AOK
